### DISCUSSION TITLE
MSE Loss function with SemanticSegmentationTask

### DISCUSSION THREAD

**Discussion by ghost** [torchgeo/torchgeo#1415] [Q&A] [answered]
Is there a way to run SemanticSegmentationTask with RMSE or MSE loss with torchgeo? Currently, supports 'ce', 'jaccard' or 'focal' loss.

**Comment by isaaccorley** [comment]
We have `torchgeo.trainers.PixelwiseRegressionTask` specifically for this. Let me know if you encounter any issues.

**Reply by adamjstewart** [reply, marked as answer]
We should think about ways to more easily integrate all metrics in case someone wants a different classification metric. Currently the only way to do this is something like:
```python
trainer = SemanticSegmentationTask(...)
trainer.train_metrics = MetricCollection(...)
trainer.val_metrics = ...
trainer.test_metrics = ...
```

**Reply by ghost** [reply]
Thanks very much for your reply. torchgeo.trainers.PixelwiseRegressionTask seems what I need for my project. But is not available for me. I am using torchgeo version 0.3.1. Also, is there a notebook that shows end2end implementation ?

**Reply by adamjstewart** [reply]
> But is not available for me. I am using torchgeo version 0.3.1.

`PixelwiseRegressionTask` will be added in the 0.5.0 release (due sometime next month if all goes according to plan). Until then, you can install a development version. See https://torchgeo.readthedocs.io/en/stable/user/installation.html for instructions to do that.

> is there a notebook that shows end2end implementation ?

See #575
